qiskit/circuit/quantumcircuit.py

```diff
@@ -1278,16 +1352,82 @@ def layout(self) -> Optional[TranspileLayout]:
         selected physical qubits on the :class:`~.Target`, and a final layout
         which is an output permutation caused by :class:`~.SwapGate`\s
         inserted during routing.
+
+        Example:
+
+            .. plot::
+                :include-source:
+                :nofigs:
+
+                from qiskit import QuantumCircuit
+                from qiskit.circuit.library import GroverOperator, Diagonal
+                from qiskit.providers.fake_provider import GenericBackendV2
+                from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
+
+                # Create circuit to test transpiler on
+                oracle = Diagonal([1] * 7 + [-1])
+                qc = QuantumCircuit(3)
+                qc.h([0, 1, 2])
+                qc = qc.compose(GroverOperator(oracle))
```

Comment:
Nice example! Just a suggestion for a simpler circuit
```suggestion
                from qiskit.providers.fake_provider import GenericBackendV2
                from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager

                # Create circuit to test transpiler on
                qc = QuantumCircuit(3)
                qc.h([0, 1, 2])
```